tests/sentry/workflow_engine/endpoints/test_organization_detector_types.py

```diff
@@ -0,0 +1,28 @@
+from sentry.incidents.grouptype import MetricAlertFire
+from sentry.issues.grouptype import MonitorIncidentType, UptimeDomainCheckFailure
+from sentry.testutils.cases import APITestCase
+from sentry.testutils.silo import region_silo_test
+
+
+class OrganizationDataConditionAPITestCase(APITestCase):
+    endpoint = "sentry-api-0-organization-detector-type-index"
+
+    def setUp(self):
+        super().setUp()
+        self.login_as(user=self.user)
+
+        self.create_detector(name="metric detector", type=MetricAlertFire.slug)
+        self.create_detector(name="crons detector", type=MonitorIncidentType.slug)
+        self.create_detector(name="uptime detector", type=UptimeDomainCheckFailure.slug)
+
+
+@region_silo_test
+class OrganizationDataConditionIndexBaseTest(OrganizationDataConditionAPITestCase):
+    def test_simple(self):
+        response = self.get_success_response(self.organization.slug, status_code=200)
+        assert len(response.data) == 3
+        assert set(response.data) == {
+            MetricAlertFire.slug,
+            MonitorIncidentType.slug,
+            UptimeDomainCheckFailure.slug,
+        }
```

Comment:
Does the order matter for this result? is there a specific sort we need to think about for the UI?

Reply:
@natemoo-re Do you have any thoughts on this? My initial thought is that the API shouldn't need to worry about ordering

Reply:
This currently matches how the UI is sorted! Sort order here _could_ map directly to how the list is displayed, but if that's unintuitive, we can protect against it on the frontend by using this response like a filter.

Reply:
Since we are getting the detector types from doing a select on the Detector model, we can't guarantee that the API will return in a certain order every time unless we implement sorting here somehow

Reply:
I guess technically they are alphabetical so we could sort that way 🤔

Reply:
alphabetical makes sense to me!